### PR TITLE
EOS-17389: LR R2: Mini Provisioner - Convert s3_setup to python - config phase <patch 1: changes in setup_ldap.sh>

### PR DESCRIPTION
**Pull request by pratyush-seagate**
commit 6f6e6cfa2019ea1753a747a4470ec28253779c44
Author: pratyush-seagate <pratyush.k.khan@seagate.com>
Date:   Fri Feb 12 01:26:46 2021 -0700

        EOS-17389: LR R2: Mini Provisioner - Convert s3_setup to python - config phase <patch 1: changes in setup_ldap.sh>
        Change description:
                As described in ticket EOS-17389 for setup_ldap.sh and in caller in s3_setup
        Added/modified/deleted files:
                modified:   scripts/ldap/setup_ldap.sh
                modified:   scripts/provisioning/s3_setup
            Unit test:
                    Run "s3_setup config" with these changes and verified steps worked. Performed complete mini-provisioner s3 steps

    Signed-off-by: pratyush-seagate <pratyush.k.khan@seagate.com>

### CODE REVIEW

File: scripts/ldap/setup_ldap.sh [hunk 1]
```diff
@@ -98,22 +99,28 @@ cp -f $INSTALLDIR/olcDatabase\=\{2\}mdb.ldif /etc/openldap/slapd.d/cn\=config/
...
-    # decrypt the passwords read from the confstore
-    LDAPADMINPASS=$(s3cipher decrypt --data "$sgiamadminpassd" --key "$cipherkey")
-    ROOTDNPASSWORD=$(s3cipher decrypt --data "$rootdnpasswd" --key "$cipherkey")
-else # Fetch Root DN & IAM admin passwords from User
-    echo -en "\nEnter Password for LDAP rootDN: "
-    read -s ROOTDNPASSWORD && [[ -z $ROOTDNPASSWORD ]] && echo 'Password can not be null.' && exit 1
+if [ -z "$LDAPADMINPASS" ]
+then
+    if [[ $defaultpasswd == true ]]
+    then # Use same default password for both for empty ones
```
Comment (amitwac2608) [inline scripts/ldap/setup_ldap.sh:105]: please improve the text of the comment, its not easy to understand what the comment is trying to convey !

File: scripts/ldap/setup_ldap.sh [hunk 2]
```diff
@@ -98,22 +99,28 @@ cp -f $INSTALLDIR/olcDatabase\=\{2\}mdb.ldif /etc/openldap/slapd.d/cn\=config/
...
+        read -s LDAPADMINPASS && [[ -z $LDAPADMINPASS ]] && echo 'Password can not be null.' && exit 1
+    fi
+fi
 
-    echo -en "\nEnter Password for LDAP IAM admin: "
-    read -s LDAPADMINPASS && [[ -z $LDAPADMINPASS ]] && echo 'Password can not be null.' && exit 1
+if [ -z "$ROOTDNPASSWORD" ]
+then
+    if [[ $defaultpasswd == true ]]
+    then # Use same default password for both for empty ones
```
Comment (amitwac2608) [inline scripts/ldap/setup_ldap.sh:117]: same - improve comment text, I did not get what the comment says !

**Review by amitwac2608 — APPROVED** [review]

**Review by AjinkyaDhumal — APPROVED** [review]
Looks Good

**Comment by nileshgovande** [comment]
All jobs passed. Merging

**Pull request merged**